project_development_tips.md

```diff
@@ -0,0 +1,293 @@
+readily consumed by other dapps through on-chain events with no new integration 
+required. With your help and participation we can further accelerate healthy and vibrant 
+network effects across the Flow ecosystem!
+
+Some of these suggestions might seem somewhat unnecessary,
+but it is important to model what a project can do to manage its smart contracts the best
+so that hopefully all of the other projects follow suit.
+
+This also assumes standard software design best practices also apply.
+Indeed, many of these suggestions are more general software design best practices,
+but there may be others that are assumed but not included here.
+
+### Implementing These Practices
+
+This document serves as mostly an outline of best practices the projects should follow.
+As with all best practices, teams will choose which applies to them and their work process,
+however, we recommend that teams explicitly define a minimum acceptable set of standards
+for themselves along with the mechanisms to ensure they are being observed.
+
+Some teams may also have their own set of development standards that acheive a similar goal
+to these. These recommendations are not meant to be the only paths to success,
+so if a team disagrees with some of these and wants to do things their own way,
+they are welcome to pursue that. This document just shows some generic suggestions
+for teams who might not know how they want to manage their project.
+
+## Design Process
+
+Smart contracts usually manage a lot of value, have many users, and are difficult to upgrade
+for a variety of reasons. Therefore, it is important to have a clearly defined design
+process for the smart contracts before much code is written so that the team
+can set themselves up for success.
+
+Here are some recommendations for how projects can organize the foundations of their projects.
+
+### Projects should ensure that there is strong technical leadership for ther smart contracts
+
+Developing a dapp requires a clear vision for the role of the smart contract and how it's integrated.
+Security vulnerabilities may arise from bugs directly in smart contract code (and elsewhere in the system).
+Asynchronous interaction vectors may lead to forms of malicious abuse,
+DOS etc in a contract triggering explosive gas costs for the developer or other problems.
+
+We recommend that engineers leading a project and deploying to mainnet have an understanding
+of software and security engineering fundamentals and have been thorough
+in their Cadence skills development. More in-depth resources for learning Cadence
+are available [here](https://developers.flow.com/cadence).
+
+The technical leader should be someone who understands Cadence well and has written Cadence smart contracts
+before. Production-level smart contracts are not the place for beginners to get their start.
+
+It should be this person’s responsibility to lead design discussions
```

Comment:
Do we want to provide more guidance on what should be covered in the design documents? Like, motivation, actual design, pros and cons of different possible designs, potential issues and how they will be mitigated, scalability, rollout process, monitoring during rollout and after etc.